### PR TITLE
DR-306: kubernetes deploy

### PR DESCRIPTION
**Pull request by ahaessly**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,40 +3,68 @@ The repo for the terra data repository built by the jade team.
...
-### Google project and account
+## Create kubernetes cluster
+
+In the google cloud console, within your personal project, go to Kubernetes Engine -> Clusters and create a cluster. In the default pool section change nodes to 1 and machine type to small. Open the Advanced section and click the checkbox to enable VPC native. Then click create.
+
+Once your cluster has finished creating, click the Connect button next to you cluster info. Copy the command and execute it on your local system. Now, if you click on docker -> kubernetes you should see a check next to the cluster you just created.
+
+
+## Create a service account
+In the google cloud console, go to IAM & Admin
```
Comment (ddietterich) [inline README.md]: I _**really, really**_ appreciate this documentation!

On this step, what permissions does the service account need? What will it be used for?

File: README.md [hunk 2]
```diff
@@ -3,40 +3,68 @@ The repo for the terra data repository built by the jade team.
...
+## Deploying to kubernetes
+### Deploying in your own test account (not dev, integration, etc)
+#### Environment variables
+    GOOGLE_CLOUD_PROJECT
+    ENVIRONMENT (local, dev)
+    
+    
+Add the service account key for your project to vault:
+
+    vault write secret/dsde/firecloud/local/datarepo/sa-key{project}.json @<localfilename>.json
```
Comment (ddietterich) [inline README.md]: There is probably something I need to read before this that would tell me to install vault on my system and set it up some way?

File: README.md [hunk 3]
```diff
@@ -3,40 +3,68 @@ The repo for the terra data repository built by the jade team.
...
 
     gcloud config set project <project-name>
     
-or specify your project in the environment variable: `GOOGLE_CLOUD_PROJECT`.
-
+    
 To see what you currently have set, use: `gcloud config list`
 
-### OIDC Environment variables
+When running locally, we are not using the proxy. Therefore, the system doesn't know your user email. Edit the `src/main/resources/application.properties` file and set the userEmail field. If you are running sam locally, set `sam.basePath` to `https://local.broadinstitute.org:50443`.
```
Comment (ddietterich) [inline README.md:59]: Can you set the property in an environment variable, instead?

**Review by ddietterich — COMMENTED** [review]
I have a few comments on the (very helpful) documentation. Otherwise, it looks fine as far as I can see. That isn't too far, since I don't grasp the context of the work.

What will happen to my environment if this gets merged before our environment alignment on Tuesday?

File: README.md (hunk 1 above)
Comment (RoriCremer) [reply README.md]: yes! This is amazing!

File: README.md (hunk 2 above)
Comment (ahaessly) [reply README.md]: i'd be surprised if you don't already have it. i'm not sure exactly where i got it.

File: README.md (hunk 3 above)
Comment (ahaessly) [reply README.md:59]: not currently. we use the spring config mechanism to get the url. and we have to have it for the repo to work. we will probably want to ctmpl this file though

**Comment by ahaessly** [comment]
@ddietterich you will need to have registered a dev email with firecloud (i thought we all did this already though - maybe i'm wrong. have you been logging into swagger when you use it?)
then set that email in the application.properties file. you will need to use new new values for OAUTH client id and secret. then you should be able to run locally without issue. the rest of the PR is for when you want to set up kubernetes in your project.

**Pull request merged**